airflow/api_connexion/endpoints/pool_endpoint.py

```diff
@@ -26,18 +30,33 @@ def delete_pool():
     raise NotImplementedError("Not implemented yet.")
 
 
-def get_pool():
+@provide_session
+def get_pool(pool_name, session):
     """
     Get a pool
     """
-    raise NotImplementedError("Not implemented yet.")
+    pool_id = pool_name
+    query = session.query(Pool)
+    pool = query.filter(Pool.pool == pool_id).one_or_none()
+
+    if pool is None:
+        raise NotFound("Pool not found")
+    return pool_schema.dump(pool)
 
 
-def get_pools():
+@provide_session
+def get_pools(session):
     """
     Get all pools
     """
-    raise NotImplementedError("Not implemented yet.")
+    offset = request.args.get(parameters.page_offset, 0)
+    limit = min(int(request.args.get(parameters.page_limit, 100)), 100)
+
+    query = session.query(Pool)
+    query = query.offset(offset).limit(limit)
+
+    pools = query.all()
+    return pool_collection_schema.dump(PoolCollection(pools=pools, total_entries=len(pools))).data
```

Comment:
Total entries should contain the number of all items, not just the number of items on the current page.

Reply:
Thanks.
Here is the change `ce9b79f`